libs/application/template-api-modules/src/lib/modules/templates/health-insurance/health-insurance.utils.ts

```diff
@@ -0,0 +1,116 @@
+  /*
+   * Convert userStatus:
+   * employed: 'O'
+   * pensioner: 'P'
+   * student: 'S'
+   * other: 'O'
+   */
+  let userStatus = ''
+  switch (extractAnswer(application.answers, 'status.type')) {
+    case 'pensioner':
+      userStatus = 'P'
+      break
+    case 'student':
+      userStatus = 'S'
+      break
+    default:
+      userStatus = 'O'
+      break
+  }
+
+  // Extract attachments
+  const arrFiles: string[] = Object.keys(application.attachments) ?? []
+  if (userStatus == 'S' && arrFiles.length <= 0) {
+    logger.error(
+      `Student's application must have confirmation of student document`,
+    )
+    throw new Error(
+      `Student's application must have confirmation of student document`,
+    )
+  }
+
+  // Extract national registry date
+  // It could be null
+  const registryDate = extractAnswer(
+    application.externalData,
+    'nationalRegistry.data.lastUpdated',
+  )
+  // if (!registryDate){
+  //   throw new Error(
+  //     `Could not find registry date`,
+  //   )
+  // }
+
+  return {
+    applicationNumber: application.id,
+    applicationDate: application.modified,
+    nationalId: application.applicant,
+    foreignNationalId:
+      extractAnswer(application.answers, 'formerInsurance.personalId') ?? '',
+    name: extractAnswer(application.answers, 'applicant.name') ?? '',
+    address:
+      extractAnswer(application.answers, 'applicant.address') ?? undefined,
+    postalAddress:
+      extractAnswer(application.answers, 'applicant.postalCode') ?? undefined,
+    citizenship:
+      extractAnswerFromJson(application.answers, 'applicant.citizenship')
+        .name ?? undefined,
+    email: extractAnswer(application.answers, 'applicant.email') ?? '',
+    phoneNumber:
+      extractAnswer(application.answers, 'applicant.phoneNumber') ?? '',
+    // IF registry date is null then used applicationDate insteads for now
+    // TODO: talk with Program
+    residenceDateFromNationalRegistry:
+      registryDate instanceof Date
+        ? new Date(registryDate)
+        : application.modified,
+    residenceDateUserThink:
+      registryDate instanceof Date
+        ? new Date(registryDate)
+        : application.modified,
+    userStatus: userStatus,
+    isChildrenFollowed:
+      extractAnswer(application.answers, 'children') == 'no' ? 0 : 1,
+    previousCountry:
+      extractAnswerFromJson(application.answers, 'formerInsurance.country')
+        .name ?? '',
+    previousCountryCode:
+      extractAnswerFromJson(application.answers, 'formerInsurance.country')
+        .countryCode ?? '',
+    previousIssuingInstitution:
+      extractAnswer(application.answers, 'formerInsurance.institution') ?? '',
+    isHealthInsuredInPreviousCountry:
+      extractAnswer(application.answers, 'formerInsurance.entitlement') == 'yes'
+        ? 1
+        : 0,
+    additionalInformation:
+      extractAnswer(application.answers, 'additionalRemarks') ?? '',
```

Comment:
According to Programm specification I think this should be 
`extractAnswer(application.answers, 'formerInsurance.entitlementReason') ?? ''`
And we should add another key to the return object
`comment: extractAnswer(application.answers, 'additionalRemarks') ?? '',`

Reply:
We dont have comment tag in Program  call. There is only additionalInformation ( which is the comment in latest page ). So basicly, we dont use additional comment when Student added with confirmation document.

Which one should I use for additionalInformation?

Reply:
merged them together and send